dev/run-tests.py

```diff
@@ -82,6 +82,7 @@ def identify_changed_files_from_git_commits(patch_sha, target_branch=None, targe
         run_cmd(['git', 'fetch', 'origin', str(target_branch+':'+target_branch)])
     else:
         diff_target = target_ref
+    run_cmd(['git', 'rebase', diff_target])
```

Comment:
This is going to have side-effects, which the current method does not. I wonder whether we can fix this in a different way by using a different `patch_sha` which is post-rebase rather than pushing rebasing logic into here.